.github/workflows/autoware-base.yaml

```diff
@@ -0,0 +1,29 @@
+name: autoware-base
+
+on:
+  schedule:
+    - cron: 0 0 15 * * # every 15th of the month
+  workflow_dispatch:
+
+jobs:
+  load-env:
+    uses: ./.github/workflows/load-env.yaml
+
+  docker-build-and-push-base:
+    needs: load-env
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out this repository
+        uses: actions/checkout@v4
+
+      - name: Set up QEMU
```

Comment:
I have noticed that you introduced QEMU to build arm image.

I think it's good to do it so that we won't have to switch runners to create images, but have you already tested if this might cause any issue when running on the actual arm system? It's okay to do the test once we have actual working autoware image in the later PRs, but I wanted to know how much tests you have done at this point.

Reply:
I have confirmed that the `core` packages can be built using `colcon build`. In the next PR, I also want to conduct tests, such as checking whether `ros2 launch` works when integrated with `docker/Dockerfile`.

```
$ docker run --rm -it -v $PWD/src/core:/autoware/src/core ghcr.io/youtalk/autoware-base:latest
$ colcon build
Starting >>> autoware_lint_common
Starting >>> autoware_planning_msgs
Starting >>> autoware_map_msgs
Starting >>> autoware_adapi_v1_msgs
Starting >>> autoware_adapi_version_msgs
Starting >>> autoware_common_msgs
Starting >>> autoware_control_msgs
Starting >>> autoware_internal_msgs
Starting >>> autoware_localization_msgs
Starting >>> autoware_perception_msgs
Starting >>> autoware_sensing_msgs
Finished <<< autoware_lint_common [1.60s]
Starting >>> autoware_cmake
...
```

Note that since `docker/Dockerfile.base` does not involve compilation, QEMU can be used. However, for `docker/Dockerfile`, it cannot be used as the build time increases several-fold.